### ISSUE TITLE
Installation getting failed on fresh machine

### ISSUE THREAD

**Issue by vpsinghbaghel** [litespeedtech/cyberpanel#707]
Platform: DigitalOcean Droplet
OS: CentOS 8 
Fresh Installation.
Issue: Getting following error while installing Cyberpanel on a fresh VPS:
Today only getting this issue. Installed Cyberpanel on several fresh VPS on CentOS 8 on DigitalOcean and other platforms.

""

Resolving get.acme.sh (get.acme.sh)... 2606:4700:3032::6815:223e, 2606:4700:3031::ac43:c710, 104.21.34.62, .                                                            ..
Connecting to get.acme.sh (get.acme.sh)|2606:4700:3032::6815:223e|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘STDOUT’

-                              [ <=>                                     ]     253  --.-KB/s    in 0s

2021-08-12 13:11:54 (9.13 MB/s) - written to stdout [253]

sh: line 1: syntax error near unexpected token `<'
sh: line 1: `<meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0"/><style>                                                            body{margin:0px;padding:0px;}iframe{width:100%;height:100%}</style><iframe src="https://www.airtel.in/dot/"                                                             width="100%" height="100%" frameborder=0></iframe>'

[08.12.2021_13-11-54] #########################################################################

[08.12.2021_13-11-54] Running: /root/.acme.sh/acme.sh --upgrade --auto-upgrade

[08.12.2021_13-11-54] #########################################################################

Traceback (most recent call last):
  File "install.py", line 2369, in <module>
    main()
  File "install.py", line 2283, in main
    checks.installAcme()
  File "install.py", line 2101, in installAcme
    preFlightsChecks.call(command, self.distro, command, command, 1, 0, os.EX_OSERR)
  File "install.py", line 219, in call
    res = subprocess.call(shlex.split(command))
  File "/usr/lib64/python3.6/subprocess.py", line 287, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib64/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/root/.acme.sh/acme.sh': '/root/.acme.sh/acme.sh'
Oops, something went wrong...
""

**Comment by qtwrk** [comment]
```
Resolving get.acme.sh (get.acme.sh)... 2606:4700:3032::6815:223e, 2606:4700:3031::ac43:c710, 104.21.34.62, . ..
Connecting to get.acme.sh (get.acme.sh)|2606:4700:3032::6815:223e|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘STDOUT’

                         [ <=>                                     ]     253  --.-KB/s    in 0s
2021-08-12 13:11:54 (9.13 MB/s) - written to stdout [253]

sh: line 1: syntax error near unexpected token <' sh: line 1: <style> body{margin:0px;padding:0px;}iframe{width:100%;height:100%}</style><iframe src="https://www.airtel.in/dot/" width="100%" height="100%" frameborder=0></iframe>'
```

this looks like get.acme.sh returned a HTML file instead of shell script ,  try again later on , see how it goes

**Comment by vpsinghbaghel** [comment]
Thank you for responding. I will try again tomorrow. on datacenter in india and any other too. Thank You.

**Comment by KewalT** [comment]
I have similar issue tried 3 times with multiple Ubuntu instance in Oracle cloud still no luck. The error is as below:

Running: root/.acme.sh/acme.sh --upgrade --auto-upgrade

Errno 2 No such file or directory: 'root/.acme.sh/acme.sh'
Oops, something went wrong..

**Comment by KewalT** [comment]
Please help! I am stuck and has some critical dependencies..

**Comment by vpsinghbaghel** [comment]
> Please help! I am stuck and has some critical dependencies..

Are you selecting Banglore Datacenter in Digital Ocean ?
Can you try any other datacenter for droplet and run the setup again and let us know it works or not.
Thank You.

**Comment by KewalT** [comment]
> Datacenter

I am doing it in Oracle Cloud, Mumbai data center. The issue is in CentOS & Ubuntu too, so I think the issue is with acme.sh.

Thanks!

**Comment by vpsinghbaghel** [comment]
> > Datacenter
> 
> I am doing it in Oracle Cloud, Mumbai data center. The issue is in CentOS & Ubuntu too, so I think the issue is with acme.sh.
> 
> Thanks!

I request you to please choose any datacenter outside India and provide your experience it works or not.
I am completely understanding what you are saying. I did not notice that you are using Oracle Cloud. So, I said Banglore because that's the only datacenter Digital Ocean has in India. Actually I meant selecting datacenter outside India.
I have gone through some troubleshooting for the issue since yesterday.
So, whichever cloud platform your VPS exists on, Please create a new VPS with datacenter selected anywhere outside India and see it works or not. OS doesn't matter. You can choose any OS.

**Comment by KewalT** [comment]
Thanks for your reply. After changing the data center, means preferring it outside India worked for you? I am not having great expertise in it. Please advice as I may need to change it accordingly.

**Comment by vpsinghbaghel** [comment]
> Thanks for your reply. After changing the data center, means preferring it outside India worked for you? I am not having great expertise in it. Please advice as I may need to change it accordingly.

I mean to say, when you create any VM or Instance there, You get an option to select Datacenter location and Availability Zone.
Please create a VM/Instance and select any AZ which is not located in India. Select any OS image which is supported by CyberPanel and then try installing cyberpanel on it.
I haven't tested this particular solution yet. But most probably it will work.

**Comment by istiak101** [comment]
See #705 

`wget -O acme.sh https://raw.githubusercontent.com/acmesh-official/get.acme.sh/master/index.html`
`chmod +x acme.sh`
`./acme.sh`

**Comment by KewalT** [comment]
Thanks for you help!

I did install acme.sh first as suggested and then the installation script but now got error as below:

Reading package lists... Done
Building dependency tree
Reading state information... Done
libffi-dev is already the newest version (3.3-4).
python3-dev is already the newest version (3.8.2-0ubuntu2).
build-essential is already the newest version (12.8ubuntu1.1).
libssl-dev is already the newest version (1.1.1f-1ubuntu2.5).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree
Reading state information... Done
python3-venv is already the newest version (3.8.2-0ubuntu2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree
Reading state information... Done
locales is already the newest version (2.31-0ubuntu9.2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Generating locales (this might take a while)...
  en_US.UTF-8... done
Generation complete.
Requirement already satisfied: virtualenv==16.7.9 in /usr/local/lib/python3.8/dist-packages (16.7.9)
--2021-08-14 03:37:53--  https://raw.githubusercontent.com/usmannasir/cyberpanel/v2.1.1/requirments.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1468 (1.4K) [text/plain]
Saving to: ‘/usr/local/requirments.txt’

/usr/local/requirments.txt                      100%[====================================================================================================>]   1.43K  --.-KB/s    in 0s

2021-08-14 03:37:53 (13.8 MB/s) - ‘/usr/local/requirments.txt’ saved [1468/1468]

Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /usr/local/CyberPanel/bin/python3
Traceback (most recent call last):
  File "/usr/local/bin/virtualenv", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/virtualenv.py", line 860, in main
    create_environment(
  File "/usr/local/lib/python3.8/dist-packages/virtualenv.py", line 1162, in create_environment
    install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages=site_packages, clear=clear, symlink=symlink)
  File "/usr/local/lib/python3.8/dist-packages/virtualenv.py", line 1532, in install_python
    shutil.copyfile(executable, py_executable)
  File "/usr/lib/python3.8/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
OSError: [Errno 26] Text file busy: '/usr/local/CyberPanel/bin/python3'
above command failed...

**Comment by KewalT** [comment]
PLEASE HELP ME! I am stuck since few days and I have very critical dependencies...

Platform: Oracle cloud, Mumbai server
OS: Ubuntu 20
Fresh Installation.

Now I just tried again on new instance. On fresh instance I ran these commands first : 

wget -O acme.sh https://get.acme.sh
chmod +x acme.sh
./acme.sh6. After installing acme.sh rerun the installation script - 
sh <(curl https://cyberpanel.net/install.sh || wget -O - https://cyberpanel.net/install.sh)

Then our main command - sh <(curl https://cyberpanel.net/install.sh || wget -O - https://cyberpanel.net/install.sh)

Still got below error:

[08.14.2021_04-52-16] #########################################################################

[08.14.2021_04-52-16] Running: apt-get -y install cron

[08.14.2021_04-52-16] #########################################################################

Reading package lists... Done
Building dependency tree
Reading state information... Done
cron is already the newest version (3.0pl1-136ubuntu1).
cron set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.



[08.14.2021_04-52-17] #########################################################################

[08.14.2021_04-52-17] Successfully ran: apt-get -y install cron.

[08.14.2021_04-52-17] #########################################################################




[08.14.2021_04-52-17] #########################################################################

[08.14.2021_04-52-17] Running: systemctl enable cron

[08.14.2021_04-52-17] #########################################################################

Synchronizing state of cron.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable cron



[08.14.2021_04-52-20] #########################################################################

[08.14.2021_04-52-20] Successfully ran: systemctl enable cron.

[08.14.2021_04-52-20] #########################################################################




[08.14.2021_04-52-20] #########################################################################

[08.14.2021_04-52-20] Running: systemctl start cron

[08.14.2021_04-52-20] #########################################################################




[08.14.2021_04-52-20] #########################################################################

[08.14.2021_04-52-20] Successfully ran: systemctl start cron.

[08.14.2021_04-52-20] #########################################################################




[08.14.2021_04-52-21] #########################################################################

[08.14.2021_04-52-21] Running: chmod 600 /var/spool/cron/crontabs/root

[08.14.2021_04-52-21] #########################################################################




[08.14.2021_04-52-21] #########################################################################

[08.14.2021_04-52-21] Successfully ran: chmod 600 /var/spool/cron/crontabs/root.

[08.14.2021_04-52-21] #########################################################################




[08.14.2021_04-52-21] #########################################################################

[08.14.2021_04-52-21] Running: systemctl restart cron.service

[08.14.2021_04-52-21] #########################################################################




[08.14.2021_04-52-21] #########################################################################

[08.14.2021_04-52-21] Successfully ran: systemctl restart cron.service.

[08.14.2021_04-52-21] #########################################################################




[08.14.2021_04-52-22] #########################################################################

[08.14.2021_04-52-22] Running: apt-get update -y

[08.14.2021_04-52-22] #########################################################################

Ign:1 http://rpms.litespeedtech.com/debian focal InRelease
Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:3 http://rpms.litespeedtech.com/debian focal Release
Hit:5 http://ap-mumbai-1-ad-1.clouds.archive.ubuntu.com/ubuntu focal InRelease
Hit:6 http://ap-mumbai-1-ad-1.clouds.archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:7 http://ap-mumbai-1-ad-1.clouds.archive.ubuntu.com/ubuntu focal-backports InRelease
Reading package lists... Done



[08.14.2021_04-52-28] #########################################################################

[08.14.2021_04-52-28] Successfully ran: apt-get update -y.

[08.14.2021_04-52-28] #########################################################################




[08.14.2021_04-52-28] #########################################################################

[08.14.2021_04-52-28] Running: apt-get install restic -y

[08.14.2021_04-52-28] #########################################################################

Reading package lists... Done
Building dependency tree
Reading state information... Done
Suggested packages:
  libjs-jquery libjs-underscore
The following NEW packages will be installed:
  restic
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 8,524 kB of archives.
After this operation, 25.7 MB of additional disk space will be used.
Get:1 http://ap-mumbai-1-ad-1.clouds.archive.ubuntu.com/ubuntu focal/universe amd64 restic amd64 0.9.6+ds-2 [8,524 kB]
Fetched 8,524 kB in 4s (2,135 kB/s)
Selecting previously unselected package restic.
(Reading database ... 87533 files and directories currently installed.)
Preparing to unpack .../restic_0.9.6+ds-2_amd64.deb ...
Unpacking restic (0.9.6+ds-2) ...
Setting up restic (0.9.6+ds-2) ...
Processing triggers for man-db (2.9.1-1) ...



[08.14.2021_04-52-42] #########################################################################

[08.14.2021_04-52-42] Successfully ran: apt-get install restic -y.

[08.14.2021_04-52-42] #########################################################################

--2021-08-14 04:52:43--  https://get.acme.sh/
Resolving get.acme.sh (get.acme.sh)... 172.67.199.16, 104.21.34.62, 2606:4700:3032::6815:223e, ...
Connecting to get.acme.sh (get.acme.sh)|172.67.199.16|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘STDOUT’

-                                                   [<=>                                                                                                  ]       0  --.-KB/s               sh: 1: Syntax error: redirection unexpected
-                                                   [ <=>                                                                                                 ]     360  --.-KB/s    in 0s

2021-08-14 04:52:43 (23.4 MB/s) - written to stdout [360]




[08.14.2021_04-52-43] #########################################################################

[08.14.2021_04-52-43] Running: /root/.acme.sh/acme.sh --upgrade --auto-upgrade

[08.14.2021_04-52-43] #########################################################################

Traceback (most recent call last):
  File "install.py", line 2357, in <module>
    main()
  File "install.py", line 2271, in main
    checks.installAcme()
  File "install.py", line 2089, in installAcme
    preFlightsChecks.call(command, self.distro, command, command, 1, 0, os.EX_OSERR)
  File "install.py", line 219, in call
    res = subprocess.call(shlex.split(command))
  File "/usr/lib/python3.8/subprocess.py", line 340, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/root/.acme.sh/acme.sh'
Oops, something went wrong...
root@o

**Comment by istiak101** [comment]
@KewalT please do not open any new issue. A workaround was already provided. If you can not do this yourself hire a sysadmin or buy CyberPanel premium support if this is very important to you.

**Comment by KewalT** [comment]
As mentioned earlier the work around provided didn't solve the issue. I already followed those steps and still the issue persisted. 

This is not related to previous or existing of any tickets hence created a new ticked.

Here I am insisting on the solution, so kindly keep the new ticket open till this issue is resolved. I am definitely open to all the suggestions to solve the issue.

**Comment by KewalT** [comment]
Reopen #709 and keep it open till the work around works..

**Comment by istiak101** [comment]
From your comments I can see that you didn't applied the correct workaround. Also I mentioned to try this on fresh instance. All current acme.sh related issues should be posted on this issue. Do not create any new issue. As Indian govt decided to block acme.sh domain this is out of our scope.

**Comment by istiak101** [comment]
I will not reopen #709. Keep all similar things in one issue.

**Comment by KewalT** [comment]
As I understand I need to move my deployment outside India and then first install 
''wget -O acme.sh https://get.acme.sh
chmod +x acme.sh
./acme.sh''

Then our main command - sh <(curl https://cyberpanel.net/install.sh || wget -O - https://cyberpanel.net/install.sh)

Please clarify if this understanding is correct?

**Comment by KewalT** [comment]
> instance

This was done in fresh instance only.

**Comment by KewalT** [comment]
> Also I mentioned to try this on fresh instance

My previous error/issue posted around an hour back was on fresh new instance only.

**Comment by istiak101** [comment]
If you deployment is outside India and you still get error try #705. If your deployment is in India try above solution in this issue #707. In both case try on fresh installation and run the main install script after successful install. And don't get hyped. This is an opensource community and someone will answer your query sooner or later, keep patience. Only open another issue if you have installed acme successfully but still your installation is failing for some other reason.

**Comment by istiak101** [comment]
> As I understand I need to move my deployment outside India and then first install
> ''wget -O acme.sh https://get.acme.sh
> chmod +x acme.sh
> ./acme.sh''
> 
> Then our main command - sh <(curl https://cyberpanel.net/install.sh || wget -O - https://cyberpanel.net/install.sh)
> 
> Please clarify if this understanding is correct?

Read my above comment.



> > Also I mentioned to try this on fresh instance
> 
> My previous error/issue posted around an hour back was on fresh new instance only.

You still tried get.acme.sh which is blocked by Indian govt. Please read carefully and don't get hyped up.

**Comment by vpsinghbaghel** [comment]
> See #705
> 
> `wget -O acme.sh https://raw.githubusercontent.com/acmesh-official/get.acme.sh/master/index.html`
> `chmod +x acme.sh`
> `./acme.sh`

@istiak101  Thank you for taking care of the issue and providing us workaround. I am registering a New Digital Ocean account for the tests and I will test with both Indian datacenter and outside. Will update you. Thank You for your support :)

Regards.
Ved.

**Comment by istiak101** [comment]
> > See #705
> > `wget -O acme.sh https://raw.githubusercontent.com/acmesh-official/get.acme.sh/master/index.html`
> > `chmod +x acme.sh`
> > `./acme.sh`
> 
> @istiak101 Thank you for taking care of the issue and providing us workaround. I am registering a New Digital Ocean account for the tests and I will test with both Indian datacenter and outside. Will update you. Thank You for your support :)
> 
> Regards.
> Ved.

Someone on Facebook already tested this on Oracle cloud upon my instruction and they installed CP successfully. Let me know in this issue if this works for your or not.

**Comment by vpsinghbaghel** [comment]
Platform: DigitalOcean
Datacenter Selected: Banglore(INDIA)
OS: CentOS 8.
Fresh Installation.

@istiak101 , I have installed Cyberpanel successfully on a fresh VPS following your instructions.
I installed acme first using below commands, and then executed CyberPanel installation script.

wget -O acme.sh https://raw.githubusercontent.com/acmesh-official/get.acme.sh/master/index.html
chmod +x acme.sh
./acme.sh

Thank you.
I will try setting up on a VPS outside India without pre-installing acme.

Regards.
Ved.

**Comment by KewalT** [comment]
**@istiak101** Thanks a TON for your help in suggesting the workaround. It worked for server in India.

**@vpsinghbaghel** Thanks a lot for confirming it worked well so even I could do it, else I was thinking to go for other panel.


Regards,
Kewal

**Comment by vpsinghbaghel** [comment]
Platform: DigitalOcean
**Datacenter Selected: San Francisco**
OS: CentOS 8.
Fresh Installation.

@istiak101 , 
I have successfully installed CyberPanel on a Fresh VPS running on datacenter outside India (San Francisco) **without pre-installing acme tool.**

Thank you.
Regards.
Ved.

**Comment by istiak101** [comment]
As this is resolved with the workaround I am closing this issue.
